### PR TITLE
Add basic Java sample app

### PR DESCRIPTION
**Pull request by damemi**

### CODE REVIEW

**Comment by damemi** [comment]
cc @jsuereth, this is your Java sample app that we were working with before --- I didn't touch any code or files, just copied everything and packaged it to build and run here. please take a look that I didn't do anything weird

File: sample-apps/java/gradlew [hunk 1]
```diff
@@ -0,0 +1,234 @@
+#!/bin/sh
+
+#
+# Copyright © 2015-2021 the original authors.
```
Comment (dashpole) [inline sample-apps/java/gradlew:4]: Is this a copyright we can include?

File: sample-apps/java/gradlew.bat [hunk 2]
```diff
@@ -0,0 +1,89 @@
+@rem
+@rem Copyright 2015 the original author or authors.
```
Comment (dashpole) [inline sample-apps/java/gradlew.bat:2]: same question on copyright.

**Review by dashpole — COMMENTED** [review]
I don't really know about the gradle stuff, but it seems good otherwise.

File: sample-apps/java/gradlew (hunk 1 above)
Comment (damemi) [reply sample-apps/java/gradlew:4]: Good question for both of these, it is Apache 2.0 so I think it's okay to use, and we would need to keep the original header to use it. Pretty sure these gradle wrappers are normally generated, so we could also switch to have steps for users to generate them themselves

**Review by dashpole — APPROVED** [review]

**Pull request merged**